### PR TITLE
feat(payment): PI-532 Refactor Adyenv2/v3 for new payment integration API

### PR DESCRIPTION
**Pull request by bc-ania**
## What?
Move Adyenv2 and Adyenv3 to a separate package

## Why?
To use the new checkout payment integration API

## Testing / Proof

Screenshot 2023-08-22 at 14 16 14

https://github.com/bigcommerce/checkout-js/assets/130754705/a83dda16-20ad-49df-9503-a7267913c1f1

@bigcommerce/checkout

### CODE REVIEW

File: packages/adyen-integration/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export { default as AdyenV2PaymentMethod } from '../src/adyenv2/AdyenV2PaymentMethod';
+export { default as AdyenV2Form } from '../src/adyenv2/AdyenV2Form';
+export { default as AdyenV2CardValidation } from '../src/adyenv2/AdyenV2CardValidation';
```
Comment (serhii-tkachenko) [inline packages/adyen-integration/src/index.ts:3]: Why do making export `AdyenV2Form`, `AdyenV2CardValidation`, `AdyenV3Form` and `AdyenV3CardValidation` components in this file?

File: packages/adyen-integration/src/adyenv3/AdyenV3PaymentMethod.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    };
+
+    const onBeforeLoad = useCallback((shopperInteraction: boolean) => {
+        ref.current.shouldShowModal = shopperInteraction;
+
+        if (ref.current.shouldShowModal) {
+            setShowAdditionalActionContent(true);
+        } else {
+            setShowAdditionalActionContent(false);
+        }
```
Comment (serhii-tkachenko) [inline packages/adyen-integration/src/adyenv3/AdyenV3PaymentMethod.tsx:72]: This code can be simplified:

```suggestion
            setShowAdditionalActionContent(ref.current.shouldShowModal);
```

File: packages/adyen-integration/src/adyenv3/AdyenV3PaymentMethod.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            additionalActionContainerId,
+            adyenOptions,
+            onBeforeLoad,
+            onComplete,
+            onLoad,
+            checkoutService,
+            checkoutState,
+        ],
+    );
+    const validateInstrument = (
```
Comment (serhii-tkachenko) [inline packages/adyen-integration/src/adyenv3/AdyenV3PaymentMethod.tsx:132]: Add an empty line above, please

File: packages/adyen-integration/src/adyenv3/AdyenV3Form.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    const instruments = checkoutState.data.getInstruments(method) || [];
+    const isInstrumentFeatureAvailable =
+        !customer?.isGuest && Boolean(method.config.isVaultingEnabled);
+
+    return (
+        <>
+            <HostedWidgetPaymentComponent
+                {...rest}
+                containerId={containerId}
+                deinitializePayment={checkoutService.deinitializePayment}
```
Comment (serhii-tkachenko) [inline packages/adyen-integration/src/adyenv3/AdyenV3Form.tsx:34]: As far as I see, `deinitializePayment` method was provided to props:
https://github.com/bigcommerce/checkout-js/pull/1433/files#diff-bd5fbd171f4dbccef385dad48890f89a0dd218f9db63bc0248657f20bac99789R184

File: packages/adyen-integration/src/adyenv3/AdyenV3Form.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import React, { FunctionComponent } from 'react';
+
+import { HostedWidgetPaymentComponent } from '@bigcommerce/checkout/hosted-widget-integration';
+import { usePaymentFormContext } from '@bigcommerce/checkout/payment-integration-api';
+import { Modal } from '@bigcommerce/checkout/ui';
+
+const AdyenV3Form: FunctionComponent<any> = ({
```
Comment (serhii-tkachenko) [inline packages/adyen-integration/src/adyenv3/AdyenV3Form.tsx:7]: We should avoid using `any` type in the project

File: packages/adyen-integration/src/adyenv2/AdyenV2Form.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
+import { createCheckoutService } from '@bigcommerce/checkout-sdk';
+import React, { FunctionComponent } from 'react';
+
+import { HostedWidgetPaymentComponent } from '@bigcommerce/checkout/hosted-widget-integration';
+import { useCheckout, usePaymentFormContext } from '@bigcommerce/checkout/payment-integration-api';
+import { Modal } from '@bigcommerce/checkout/ui';
+
+const AdyenV2Form: FunctionComponent<any> = ({
```
Comment (serhii-tkachenko) [inline packages/adyen-integration/src/adyenv2/AdyenV2Form.tsx:8]: We should avoid using `any` type

**Review by vitalii-koshovyi — APPROVED** [review]

**Review by serhii-tkachenko — APPROVED** [review]

**Comment by animesh1987** [comment]
Hey @bc-ania can you please add some screenshots as part of proof for this PR.

**Comment by bc-ania** [comment]
> 

@animesh1987 screenshot and video added

File: packages/adyen-integration/README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# adyenv2-integration
+
+This library was generated with [Nx](https://nx.dev).
```
Comment (animesh1987) [inline packages/adyen-integration/README.md:3]: Can you please add some more detail compared to the boilerplate here.

File: packages/adyen-integration/src/adyenv2/AdyenV2PaymentMethod.spec.tsx [hunk 8]
```diff
@@ -2,26 +2,35 @@ import {
     CheckoutSelectors,
     CheckoutService,
     createCheckoutService,
+    createLanguageService,
```
Comment (animesh1987) [inline packages/adyen-integration/src/adyenv2/AdyenV2PaymentMethod.spec.tsx:5]: Can we please write tests using RTL here.

File: packages/adyen-integration/src/adyenv3/AdyenV3PaymentMethod.spec.tsx [hunk 9]
```diff
@@ -2,26 +2,35 @@ import {
     CheckoutSelectors,
     CheckoutService,
     createCheckoutService,
+    createLanguageService,
     PaymentMethod,
 } from '@bigcommerce/checkout-sdk';
-import { mount, ReactWrapper } from 'enzyme';
+import { render } from '@testing-library/react';
```
Comment (animesh1987) [inline packages/adyen-integration/src/adyenv3/AdyenV3PaymentMethod.spec.tsx:8]: Same as above to use RTL

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -39,10 +39,7 @@
 # kyiv payments
 /packages/bluesnap-direct-integration @bigcommerce/kyiv-payments-team
 /packages/core/customer/checkoutSuggestion @bigcommerce/kyiv-payments-team
-/packages/core/payment/paymentMethod/AdyenV2CardValidation.tsx @bigcommerce/kyiv-payments-team
-/packages/core/payment/paymentMethod/AdyenV3CardValidation.tsx @bigcommerce/kyiv-payments-team
-/packages/core/payment/paymentMethod/AdyenV2PaymentMethod.tsx @bigcommerce/kyiv-payments-team
-/packages/core/payment/paymentMethod/AdyenV3PaymentMethod.tsx @bigcommerce/kyiv-payments-team
+/packages/adyen-integration @bigcommerce/kyiv-payments-team
```
Comment (bc-peng) [inline .github/CODEOWNERS:42]: Could you please sort the list?

File: packages/adyen-integration/project.json [hunk 11]
```diff
@@ -0,0 +1,23 @@
+{
+  "root": "packages/adyen-integration",
+  "sourceRoot": "packages/adyen-integration/src",
+  "projectType": "library",
+  "targets": {
+    "lint": {
+      "executor": "@nrwl/linter:eslint",
+      "outputs": ["{options.outputFile}"],
+      "options": {
+        "lintFilePatterns": ["packages/adyen-integration/**/*.ts"]
```
Comment (bc-peng) [inline packages/adyen-integration/project.json:10]:
```suggestion
        "lintFilePatterns": ["packages/adyen-integration/**/*.{ts,tsx}"]
```

File: packages/adyen-integration/tsconfig.json [hunk 12]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "../../tsconfig.base.json",
+  "references": [
+    {
+      "path": "./tsconfig.spec.json"
+    }
+  ],
+}
```
Comment (bc-peng) [inline packages/adyen-integration/tsconfig.json:8]: Could you please add one more line here?

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**